vue/index.js

```diff
@@ -175,3 +175,30 @@ export let ChannelErrors = {
     }
   }
 }
+
+export function useAuth(client) {
+  client = client || useClient()
+  let userId = ref(client.options.userId)
+  let isAuthenticated = ref(false)
+
+  client.on('user', newUserId => {
+    userId.value = newUserId
+  })
+
+  client.node.on('state', () => {
+    if (client.node.state === 'synchronized') {
+      isAuthenticated.value = true
```

Comment:
Let’s unbind listener on first `synchronized`. We can bind listener again on `wrong-credentials`.

Reply:
Cool idea 🤘🏻